File: PracticeSet-1/Arrays And Strings/FirstUniqueChar.java
/*
s = "leetcode"
return 0.

s = "loveleetcode",
return 2.
*/
class Solution {
    public int firstUniqChar(String s) {
        if(s.length()==0)
        {
        	return -1;
        }
        int[] charCount = new int[26];
        for(int i=0;i<s.length();i++)
        {
        	charCount[(int)s.charAt(i)-'a']++;
        }
        for(int i=0;i<s.length();i++)
        {
        	if(charCount[(int)s.charAt(i)-'a']==1)
        	{
        		return i;
        	}
        }
        return -1;
    }
}

File: CSharp/Easy/Strings/FirstUniqueChar.cs
/*
Examples:

s = "leetcode"
return 0.

s = "loveleetcode",
return 2.
*/

public class Solution {
    public int FirstUniqChar(string s) {
        int[] countArr = new int[26];
        for(int i=0;i<s.Length;i++)
        {
        	countArr[s[i]-'a']++;
        }
        for(int i=0;i<s.Length;i++)
        {
        	if(countArr[s[i]-'a']==1)
        	{
        		return i;
        	}
        }
        return -1;
    }
}